src/Compilers/CSharp/Test/Semantic/FlowAnalysis/RegionAnalysisTests.cs

```diff
@@ -4563,6 +4563,32 @@ public static void Main()
             Assert.Equal(null, GetSymbolNamesJoined(dataFlowAnalysisResults.VariablesDeclared));
         }
 
+        [WorkItem(1291, "https://github.com/dotnet/roslyn/issues/1291")]
+        [Fact]
+        public void CaptureInQuery()
+        {
+            var analysisResults = CompileAndAnalyzeDataFlowExpression(@"
+using System.Linq;
+
+public class Test
+{
+    public static void Main(int[] data)
+    {
+        int y = 1;
+        {
+            int x = 2;
+            var f2 = from a in data select a + y;
+            var f3 = from a in data where x > 0 select a;
+            var f4 = from a in data let b = 1 select /*<bind>*/a + b/*</bind>*/;
+            var f5 = from c in data where M(() => c) select c;
+        }
+    }
+    private static bool M(Func<int> f) => true;
+}");
+            var dataFlowAnalysisResults = analysisResults;
+            Assert.Equal("y, x, c", GetSymbolNamesJoined(dataFlowAnalysisResults.Captured));
```

Comment:
@tmat Please chime in on this topic. How do you determine *which* instance of a range variable is captured for EnC. Or perhaps you don't need to?